### PR TITLE
 Promote modules iam_role and iam_role_info

### PR DESCRIPTION
**Pull request by abikouo**
SUMMARY

Migrate modules iam_role and iam_role_info from community.aws
Apply deprecations:

    iam_role: module returns role information into iam_role only
    iam_role and iam_role_info: keys from returned property assume_role_policy_document are no longer snake-cased

ISSUE TYPE

    New Module Pull Request

COMPONENT NAME

iam_role
iam_role_info

### CODE REVIEW

File: plugins/modules/iam_role.py [hunk 1]
```diff
@@ -0,0 +1,769 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: iam_role
+version_added: 7.0.0
```
Comment (tremble) [inline plugins/modules/iam_role.py:10]:
```suggestion
version_added: 1.0.0
version_added_collection: community.aws
```

File: plugins/modules/iam_role.py [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+short_description: Manage AWS IAM roles
+description:
+  - Manage AWS IAM roles.
+  - This module was originally added to C(community.aws) in release 1.0.0.
```
Comment (tremble) [inline plugins/modules/iam_role.py:14]:
```suggestion
```

File: plugins/modules/iam_role_info.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: iam_role_info
+version_added: 7.0.0
```
Comment (tremble) [inline plugins/modules/iam_role_info.py:10]:
```suggestion
version_added: 1.0.0
version_added_collection: community.aws
```

File: plugins/modules/iam_role_info.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+short_description: Gather information on IAM roles
+description:
+    - Gathers information about IAM roles.
+    - This module was originally added to C(community.aws) in release 1.0.0.
```
Comment (tremble) [inline plugins/modules/iam_role_info.py:14]:
```suggestion
```

**Review by tremble — CHANGES_REQUESTED** [review]
Please use version_added_collection rather than tweaking the description.

**Comment by tremble** [comment]
682 Commits?  Something looks like it's gone very wonky here...

File: changelogs/fragments/20230915_migrate_iam_role_and_iam_role_info.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+breaking_changes:
```
Comment (alinabuzachis) [inline changelogs/fragments/20230915_migrate_iam_role_and_iam_role_info.yml:1]: Is it a breaking or major change?

Comment (tremble) [reply changelogs/fragments/20230915_migrate_iam_role_and_iam_role_info.yml:1]: Given that we have the redirects in place I'd argue it's either "Major" (just for the sake of highlighting it) or potentially even "Minor".

**Pull request closed without merging**